### PR TITLE
feat(utilities): add functions, tests, input/return class, docs for s…

### PR DESCRIPTION
**Pull request by yehui-h**
…ports_heat_stress_risk calculation, and bump version to 3.8.0

## Release Notes – Version 3.8.0

* **New Features**
  * Heat stress risk assessment model for sports athletes: calculates risk levels based on environmental conditions (temperature, humidity, radiation, air velocity) and sport type, providing actionable heat-stress management recommendations for safer outdoor activity participation

* **Documentation**
  * New model documentation with detailed parameters, return values, and usage examples
  * Version updated to 3.8.0 with changelog entries

### CODE REVIEW

File: pythermalcomfort/models/sports_heat_stress_risk.py [hunk 1]
```diff
@@ -79,66 +85,129 @@ def sports_heat_stress_risk(
...
+    vectorized_calc = np.vectorize(
+        _calc_risk_single_value, otypes=[float, float, float, float, str]
+    )
+    risk_levels, t_mediums, t_highs, t_extremes, recommendations = vectorized_calc(
         tdb=tdb, tr=tr, rh=rh, vr=vr, sport=sport
     )
 
-    # todo use the pmv_ppd_iso as a reference for the code implementation
+    # Convert scalar arrays back to scalars if input was scalar
+    if tdb.ndim == 0:
```
Comment (FedericoTartarini) [inline pythermalcomfort/models/sports_heat_stress_risk.py]: I do not think this is needed, please use the pmv_ppd_iso as a refrence and over there we are not doing this. However, if you think that this is a nice feature to have we should have a global function that does this conversion and then we should apply this to all the outputs of the models

File: pythermalcomfort/models/sports_heat_stress_risk.py [hunk 2]
```diff
@@ -79,66 +85,129 @@ def sports_heat_stress_risk(
...
+        recommendation=recommendations,
+    )
+
 
-    # todo below is a placeholder but the actual implementation should return a dataclass with the risk value
-    #  interpolated and the risk level thresholds Low, Medium, High, Extreme as per the online tool
-    #  https://sma-heat-policy.sydney.edu.au/
-    return risk_levels
+def _calc_risk_single_value(
+    tdb, tr, rh, vr, sport: _SportsValues
```
Comment (FedericoTartarini) [inline pythermalcomfort/models/sports_heat_stress_risk.py]: missing type annotations

File: pythermalcomfort/models/sports_heat_stress_risk.py [hunk 3]
```diff
@@ -63,10 +64,15 @@ def sports_heat_stress_risk(
...
+) -> SportsHeatStressRisk:
+    """Calculate sports heat stress risk levels based on environmental conditions and
+    sport-specific parameters.
 
-    todo explain what the function does and cite the paper DOI:10.1016/j.jsams.2025.03.006
+    This function assesses heat stress risk for athletes during outdoor sports by
+    combining environmental conditions with sport-specific metabolic rates and clothing
+    insulation. It uses the Predicted Heat Strain (PHS) model to determine threshold
+    temperatures for different risk categories (Low, Medium, High, Extreme), based on
+    the Sports Medicine Australia heat policy framework [SportsHeatStress2025]_.
```
Comment (FedericoTartarini) [inline pythermalcomfort/models/sports_heat_stress_risk.py]: please add a reference to the paper but also to the online tool and to this document https://sma.org.au/wp-content/uploads/2026/01/Extreme-Heat-Policy-2025_v1.pdf. please also make sure that you use the right names, the policy is called EXTREME HEAT RISK AND
RESPONSE GUIDELINES AND WEBTOOL (2025) while the online tool is called Sports Heat Tool https://sma-heat-policy.sydney.edu.au/

File: pythermalcomfort/models/sports_heat_stress_risk.py [hunk 4]
```diff
@@ -270,42 +343,56 @@ def calculate_threshold_core(x):
...
-    plt.xlabel("Dry Bulb Temperature (°C)")
-    plt.ylabel("Relative Humidity (%)")
-    plt.show()
+    Returns
+    -------
+    str
+        Evidence-based recommendation text for managing heat stress at the given
+        risk level.
+    """
+    if risk_level < 1.0:
```
Comment (FedericoTartarini) [inline pythermalcomfort/models/sports_heat_stress_risk.py:360]: please make sure that the text is the exact same as in the policy document, we will also need to update the onlie tool to make sure they all match. https://sma.org.au/wp-content/uploads/2026/01/Extreme-Heat-Policy-2025_v1.pdf

File: pythermalcomfort/classes_return.py [hunk 5]
```diff
@@ -1043,3 +1043,28 @@ class ScaleWindSpeedLog(AutoStrMixin):
...
     v_z2: npt.ArrayLike
+
+
+@dataclass(frozen=True, repr=False)
+class SportsHeatStressRisk(AutoStrMixin):
+    """Dataclass to represent the Sports Heat Stress Risk calculation results.
+
+    Attributes
+    ----------
+    risk_level_interpolated : float or list of floats
```
Comment (FedericoTartarini) [inline pythermalcomfort/classes_return.py:1054]: we need to explain that 0-1 is low, 1-2 is moderate, ...

File: tests/test_sports_heat_stress_risk.py [hunk 6]
```diff
@@ -1,48 +1,395 @@
...
 
 
-def test_sports_heat_stress_risk_scalar(monkeypatch):
-    t = 33
-    rh = 40
-    v = 2
-    tr = 20
-    result = sports_heat_stress_risk(tdb=t, tr=tr, rh=rh, vr=v, sport=Sports.MTB)
-    assert result == pytest.approx(0.9, rel=1e-3)
+@pytest.mark.parametrize(
```
Comment (FedericoTartarini) [inline tests/test_sports_heat_stress_risk.py:11]: please add more test and test some edge cases as well, like negative wind speeds, ...

File: pythermalcomfort/models/sports_heat_stress_risk.py [hunk 7]
```diff
@@ -79,66 +85,129 @@ def sports_heat_stress_risk(
...
+        t_extreme=t_extremes,
+        recommendation=recommendations,
+    )
+
 
-    # todo below is a placeholder but the actual implementation should return a dataclass with the risk value
-    #  interpolated and the risk level thresholds Low, Medium, High, Extreme as per the online tool
-    #  https://sma-heat-policy.sydney.edu.au/
-    return risk_levels
+def _calc_risk_single_value(
```
Comment (FedericoTartarini) [inline pythermalcomfort/models/sports_heat_stress_risk.py:176]: please make sure that the code in pythermalcomfort is the same as the one here. Please let me know if there are differences https://github.com/FedericoTartarini/paper-olympics-heat-stress-risk/blob/56cddc574fe3dd262c8d3b8abc69fda58290a840/risk_calculation/new_risk_eq_v2.py#L17

File: .bumpversion.toml [hunk 8]
```diff
@@ -1,5 +1,5 @@
 [tool.bumpversion]
-current_version = "3.7.1"
+current_version = "3.8.0"
```
Comment (FedericoTartarini) [inline .bumpversion.toml]: we will need to update this to 3.9 since 3.8 has already been used.

File: CHANGELOG.rst [hunk 9]
```diff
@@ -1,6 +1,11 @@
 Changelog
 =========
 
+3.8.0 (2025-12-05)
```
Comment (FedericoTartarini) [inline CHANGELOG.rst]: see previous comment.

File: pythermalcomfort/models/sports_heat_stress_risk.py (hunk 7 above)
Comment (yehui-h) [reply pythermalcomfort/models/sports_heat_stress_risk.py:176]: Algorithm Differences:
1. Sweat loss threshold: uses 825 g/hr (local) vs 850 g/hr (reference)
2. Medium risk minimum: uses 22°C (local) vs 23°C (reference)
3. Extreme risk minimum: uses 25°C (local) vs 26°C (reference)
4. Early return threshold: returns 0 risk below 21°C (local) vs 23°C (reference)

Implementation Enhancements (local):
- Returns interpolated decimal risk levels (0.0-3.0) instead of categorical integers (0,1,2,3)
- Provides richer output: includes threshold temperatures and recommendations, not just risk level
- Implements explicit warnings for solver failures instead of silent pass statements
- Uses type-safe dataclasses instead of dictionary access

Comment (FedericoTartarini) [reply pythermalcomfort/models/sports_heat_stress_risk.py:176]: 1. Let's use the reference values for the risk. 850, 23, 26, 23, but then keep the following
2. return decimal risk value, 1 decimal
3. Keep richer output
4. Keep warning
5. Use type-safe
